src/material/core/common-behaviors/error-state.ts

```diff
@@ -33,7 +35,12 @@ export interface HasErrorState {
   _parentFormGroup: FormGroupDirective;
   _parentForm: NgForm;
   _defaultErrorStateMatcher: ErrorStateMatcher;
+
+  // These properties are defined as per the `MatFormFieldControl` interface. Since
+  // this mixin is commonly used with custom form-field controls, we respect the
+  // properties (also with the public name they need according to `MatFormFieldControl`).
   ngControl: NgControl;
+  stateChanges: Subject<void>;
```

Comment:
This is a breaking change, and I don't its actually what we want. Before the `mixinErrorState` function automatically added a `stateChanges` property, now we're requiring users to do it instead

Reply:
Why would you expect the mixin to define the state changes property? 

It feels unexpected to me that a mixin would define that property, especially since such a subject can emit in other cases too (outside of the mixin). Here is the original discussion https://github.com/angular/components/pull/22875#discussion_r645060342

It's a good point that this is a breaking change though. I did not know the mixins were part of our public API..

Reply:
Yeah I could see extracting it, and I don't think it would be hard to land. It seemed like there was just one failing app in google3. I just assumed the breaking change had been done accidentally since this was labeled as a docs PR.

Reply:
I see; yeah it was totally an oversight. I've sent a separate PR for the non-breaking docs fix until we _could_ land this. WDYT?